### PR TITLE
feat: make `query_command` shell-agnostic

### LINKED ISSUE #18: Unreliable with non-standard shell
I like to use a non-standard shell for `vim.opt.shell` (Nushell). It happens to work on Linux, but does not on Windows (Nushell is cross-platform).

I think the issue is that `query_command` in `init.lua` is a shell script, that assumes the default shells per platform.

I would like to refactor the code to:
1. change `query_command` to a table, that always represents one command + arguments (no shell features allowed, e.g. pipes)
2. change `vim.fn.jobstart` to `vim.system`, which is not dependent on the shell option in Neovim.

Thoughts?

### PR DESCRIPTION
**Pull request by bugabinga**
fixes #18
`query_command` is now of type `table` so that `vim.fn.jobstart` does
not use `shell`.

Tested on Windows 11 with nushell.
Needs testing in:
* linux
* linux + sudo
* macos
* WSL

The code in `parse_query_response` now needs to handle a table, instead of a string, because the `query_command` cannot use pipes any more, which previously guaranteed that stdout consisted of one line.

### CODE REVIEW

File: lua/auto-dark-mode/init.lua [hunk 1]
```diff
@@ -86,25 +85,30 @@ local function init()
...
+			"AppsUseLightTheme",
+		}
 	else
 		return
 	end
 
 	if vim.fn.has("unix") ~= 0 then
 		if vim.loop.getuid() == 0 then
-			query_command = "su - $SUDO_USER -c " .. query_command
+			query_command = vim.tbl_extend("keep", { "su", "-", "$SUDO_USER", "-c" }, query_command)
```
Comment (nekowinston) [inline lua/auto-dark-mode/init.lua:111]: I'm unsure how this command will behave now.
It seems like `su` will spawn the default shell indicated in `/etc/passwd`?

**Review by nekowinston — CHANGES_REQUESTED** [review]
This PR so far (with my suggestions applied) breaks the plugin on Linux/NixOS for me.

I'll take another look on Linux and macOS tomorrow.

**Comment by f-person** [comment]
Hey @nekowinston! Does your commit fix the issues with the PR? Feel free to merge this if it works for you or close it otherwise :)

**Comment by nekowinston** [comment]
It doesn't my commit so far was a simple syntax error fix, the whole PR doesn't work for me in its current state.
I meant to get back to this, but I currently don't have much time for FOSS work... 😓

**Comment by f-person** [comment]
No worries, do as you wish whenever you wish with this one :)

**Comment by bugabinga** [comment]
I have tested this branch on Windows, WSL and Fedora 40 and those work.
Unfortunately, I have no NixOs system to test on.

**Comment by nekowinston** [comment]
Sorry for taking ages to get to this! :sweat: 
Going to try this on my other machines; will merge it should everything work well.

**Review by nekowinston — APPROVED** [review]
Successfully tested on:
- Linux/NixOS
- macOS Sequoia
- Windows 10
- WSL

and again, thanks for your contribution 😄

**Pull request merged**